_drafts/i2c-overview.md

```diff
@@ -0,0 +1,402 @@
+---
+title: "I2C in a Nutshell"
+description:
+  ""
+tag: [buses]
+author: francois
+image: 
+---
+
+<!-- excerpt start -->
+I2C is perhaps the most commonly used bus to connect ICs together. As such,
+firmwae engineers encounter it on most project. In this post, we explain how I2C
+works and go through common bugs and how to investiage them.
+<!-- excerpt end -->
+
+{:.no_toc}
+
+## Table of Contents
+
+<!-- prettier-ignore -->
+* auto-gen TOC:
+{:toc}
+
+
+## Why use I2C
+
+I2C has many advantages:
+1. It is cheap to implement, with fewer pins than SPI, and a simpler physical
+   layer than CAN (no differential signalling).
+2. It supports up to 127 devices on a bus with only two pins.
+3. It goes up to 400Kbps, which is fast enough for many human interfaces as well
+   as fan control, temperature sensing and other low-speed contol loops.
+4. Historically, it has been less onerous for manufacturers to include than some
```

Comment:
maybe: an example or such ... if theres a few in particular you are thinking of

Reply:
yes, 1-wire by Maxim (patents just expired)